### PR TITLE
strip tvOS incompatible code in tvOS builds

### LINKED ISSUE #950: TvOS compiling issues
## Description

The package is not compiling on TvOS version 3.2.0 and 3.3.0
It successfully compiles on version 3.1.1

## Package versions

- React: 16.13.1
- React Native: npm:react-native-tvos@0.63.1-4
- React Native Screens: 3.2.0 or 3.3.0

### PR DESCRIPTION
**Pull request by Kazetsukai**
## Description

Fixing compile errors on tvOS
Fixes #950

## Changes

- wraps code that won't compile on tvOS with compile-time conditionals

## Test code and steps to reproduce

Attempt to use the library as-is in a tvOS React Native project, and you will have compile errors relating to some calls not being available on tvOS.

## Checklist

- [x] Included code example that can be used to test this change
- [ ] Ensured that CI passes

### CODE REVIEW

**Review by kacperkapusciak — CHANGES_REQUESTED** [review]
The `lib/` folder isn't meant to be committed. These files are automatically generated and they shouldn't be in the repo. It's added to the [.gitignore here](https://github.com/software-mansion/react-native-screens/blob/master/.gitignore#L51) and I'm confused how and why you managed to commit it anyway.

File: ios/RNSScreenStack.m [hunk 1]
```diff
@@ -507,6 +511,7 @@ - (void)dismissOnReload
   return nil;
 }
 
+#if !TARGET_OS_TV
```
Comment (WoLewicki) [inline ios/RNSScreenStack.m:514]: Should we eliminate whole `gestureRecognizerShouldBegin` method on TVOS? Is it never used there?

Comment (Kazetsukai) [reply ios/RNSScreenStack.m:514]: I am probably misunderstanding something here, but I thought this was only used for the edge swipe recognisers defined in `setupGestureHandlers` which don't exist on tvOS. My Obj-C skills aren't the best, does removing this interact with something outside of `RNSScreenStack`?

**Comment by Kazetsukai** [comment]
Woops, was playing around with something locally, didn't mean to add lib to this, reverted.

File: ios/RNSScreenStack.m (hunk 1 above)
Comment (WoLewicki) [reply ios/RNSScreenStack.m:514]: You can see here the implementation before `setupGestureHandlers` was introduced: https://github.com/software-mansion/react-native-screens/pull/761/files#diff-1c3e13a1b0dcbaa5f7cff5f3ce12cc915fb509725faf693fb65cd23ceb49c4a9L141, so I think we should do something like
```objc
#if TARGET_OS_TV
	return YES;
#else
	// rest of the code
```

and it should start here: https://github.com/software-mansion/react-native-screens/pull/761/files#diff-1c3e13a1b0dcbaa5f7cff5f3ce12cc915fb509725faf693fb65cd23ceb49c4a9R526, but I am not sure if this is 100% correct and if this method is ever fired on `tvOS`.

Comment (WoLewicki) [reply ios/RNSScreenStack.m:514]: If we do it like that, maybe we should also resolve using `simple_push` to default transition so we don't end up with some weird behaviors on `tvOS`. We could do that by returning `nil` for `tvOS` and `RNSScreenStackAnimationSimplePush` option here: `https://github.com/software-mansion/react-native-screens/blob/a6f91b6b7ef800f87c1ccc1dea95e1ebcb5604e7/ios/RNSScreenStack.m#L503`.

**Comment by WoLewicki** [comment]
I will close this since #988 has been merged. Thanks @Kazetsukai for your contribution and I added you as a co-author of #988 :tada:

**Pull request closed without merging**